### PR TITLE
📊 demography: life tables

### PR DESCRIPTION
**Pull request by lucasrodes**
Add dataset with life tables using UN and HMD.

Having UN data allows us to have a much more complete country coverage post 1950.

### CODE REVIEW

File: etl/steps/data/garden/demography/2023-10-03/life_tables.py [hunk 1]
```diff
@@ -10,10 +10,12 @@
         - HMD contains data on fewer countries, but since 1676!
         - We therefore use UN since 1950 for all countries, and HMD prior to that. We use the same source for all countries in each time period to ensure comparability across countries.
     - Age grous:
```
Comment (paarriagadap) [inline etl/steps/data/garden/demography/2023-10-03/life_tables.py:12]: groups*

File: etl/steps/data/garden/demography/2023-10-03/life_tables.py [hunk 2]
```diff
@@ -10,10 +10,12 @@
         - HMD contains data on fewer countries, but since 1676!
```
Comment (paarriagadap) [inline etl/steps/data/garden/demography/2023-10-03/life_tables.py:10]: There is this warning for this step:
```
[warning  ] Different values of 'unit' detected among variables: {'', 'years'}
```

File: etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml [hunk 3]
```diff
@@ -59,7 +59,7 @@ tables:
       number_person_years_remaining:
         title: Number of person-years remaining
         unit: person-years
-        description_short: Number of person-years remaining after exact age x.
+        description_short: Number of person-years remaining after a given age.
```
Comment (paarriagadap) [inline etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:62]: There is a commented `description_short` below. Is that intentional?

File: etl/steps/data/garden/demography/2023-10-03/life_tables.py [hunk 4]
```diff
@@ -108,13 +119,11 @@ def combine_tables(tb_hmd: Table, tb_un: Table) -> Table:
             "country": "location",
```
Comment (paarriagadap) [inline etl/steps/data/garden/demography/2023-10-03/life_tables.py:119]: assertions above should refer to HMD

**Review by paarriagadap — APPROVED** [review]

**Comment by paarriagadap** [comment]
Looks good to me!

File: etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml (hunk 3 above)
Comment (lucasrodes) [reply etl/steps/data/garden/hmd/2023-09-19/hmd.meta.yml:62]: can remove

File: etl/steps/data/garden/demography/2023-10-03/life_tables.py (hunk 2 above)
Comment (lucasrodes) [reply etl/steps/data/garden/demography/2023-10-03/life_tables.py:10]: Thanks, fixed!

**Pull request merged**